### ISSUE TITLE
Crash in BOXAPIDataOperation::connection:didReceiveData after canceling download

### ISSUE THREAD

**Issue by pdxcodemonkey** [box/box-ios-sdk#223]
My app downloads a long list of video files.  The operation must be cancelable, so I have a cancel button on the view, and when clicked, my controller walks the download queue and calls cancel on the corresponding BOXFileDownloadRequest, then the app goes on about its business.  A good percentage of the time, this results in a crash (EXC_BAD_ACCESS) in connection:didReceiveData, apparently due to calling on an NSOutputStream that has already been closed.

Crash is on the first line in connection:didReceiveData:

```
while ([self.outputStream hasSpaceAvailable])
```

I added the following trace code to the close method to test it:

```
(void)close
{
    if (self.error.code == BOXContentSDKAPIErrorAccepted) {
        // for 202, we are going to re-enqueue so we don't want to mess with the stream.
    }
    else {
        NSLog(@"Removing %p as delegate for %p", self, _outputStream);
        self.outputStream.delegate = nil;
        [self.outputStream close];
        _outputStream = nil;
        NSLog(@"%p - output stream is nil", self);
    }
}
```

Not surprisingly, this made the crash much harder to reproduce, but I did eventually hit it.  As expected, the first trace was emitted for the object in question, but not the second, so it appears the callback is coming in on another thread after the output stream is closed, but before it's set to nil.  The output stream's delegate should already be set to nil, though - maybe a bug in NSOutputStream?  Weird stuff.

**Comment by h2kuo** [comment]
@austxcodemonkey, could you try using our current master branch of the SDK? We believe this issue should have been fixed with PRs #213 and #216, which have not yet been released to CocoaPods.

Please let us know if you're still running into any issues!

**Comment by pdxcodemonkey** [comment]
Thanks, this does appear to have fixed the problem!
